torchx/schedulers/aws_batch_scheduler.py

```diff
@@ -582,36 +595,46 @@ def list(self) -> List[ListAppResponse]:
         for resp in self._client.get_paginator("describe_job_queues").paginate():
             queue_names = [queue["jobQueueName"] for queue in resp["jobQueues"]]
             for qn in queue_names:
-                apps_in_queue = self._list_by_queue(qn)
-                all_apps += [
-                    ListAppResponse(
-                        app_id=f"{qn}:{app['jobName']}",
-                        state=JOB_STATE[app["status"]],
-                    )
-                    for app in apps_in_queue
-                ]
+                all_apps.extend(self._list_by_queue(qn))
         return all_apps
 
-    def _list_by_queue(self, queue_name: str) -> List[Dict[str, Any]]:
-        # By default only running jobs are listed by batch/boto client's list_jobs API
+    def _list_by_queue(self, queue_name: str) -> List[ListAppResponse]:
+        # By default, only running jobs are listed by batch/boto client's list_jobs API
         # When 'filters' parameter is specified, jobs with all statuses are listed
         # So use AFTER_CREATED_AT filter to list jobs in all statuses
         # milli_seconds_after_epoch can later be used to list jobs by timeframe
-        milli_seconds_after_epoch = "1"
-        job_summary_list = []
+        MS_AFTER_EPOCH = "1"
+        EVERY_STATUS = {"name": "AFTER_CREATED_AT", "values": [MS_AFTER_EPOCH]}
+
+        jobs = []
         for resp in self._client.get_paginator("list_jobs").paginate(
             jobQueue=queue_name,
-            filters=[
-                {
-                    "name": "AFTER_CREATED_AT",
-                    "values": [
-                        milli_seconds_after_epoch,
-                    ],
-                },
-            ],
+            filters=[EVERY_STATUS],
+            # describe-jobs API can take up to 100 jobIds
+            PaginationConfig={"MaxItems": 100},
         ):
-            job_summary_list.extend(resp["jobSummaryList"])
-        return job_summary_list
+
+            def get_torchx_submitted_jobs(job_ids: List[str]) -> List[ListAppResponse]:
```

Comment:
can we define this outside of the forloop since it doesn't depend on any variables in the for loop?

Reply:
done